src/tools/genpolicy/src/registry.rs

```diff
@@ -57,21 +60,20 @@ pub struct DockerRootfs {
 }
 
 /// This application's image layer properties.
-#[derive(Clone, Debug)]
+#[derive(Clone, Debug, Serialize, Deserialize)]
 pub struct ImageLayer {
     pub diff_id: String,
     pub verity_hash: String,
 }
 
 impl Container {
-    pub async fn new(config: &Config, image: &str) -> Result<Self> {
+    pub async fn new(use_cached_files: bool, image: &str) -> Result<Self> {
         info!("============================================");
         info!("Pulling manifest and config for {:?}", image);
         let reference: Reference = image.to_string().parse().unwrap();
         let auth = build_auth(&reference);
 
         let mut client = Client::new(ClientConfig {
-            protocol: ClientProtocol::HttpsExcept(config.insecure_registries.clone()),
```

Comment:
@Redent0r @danmihai1 I just noticed that you removed [support for insecure registries](https://github.com/kata-containers/kata-containers/pull/9294) here. Is this not compatible with the changed caching? Can we please reintroduce it?

Reply:
@burgerdev , sorry, that was a mistake. We'll look at adding this back.

Reply:
Adding back in https://github.com/kata-containers/kata-containers/pull/9786. Sorry about that @burgerdev !